File: 01-data-preprocessing/correlation_heatmap.r
df <- read.csv("movies.csv")
df$VOTES <- as.integer(gsub(",", "", df$VOTES))
df$Gross <- as.numeric(gsub("[$M]", "", df$Gross))
df <- df[complete.cases(df$VOTES, df$Gross), ]

numeric_df <- df[sapply(df, is.numeric)]
corr_matrix <- cor(numeric_df)

library(heatmaply)
heatmaply(corr_matrix, main = "Correlation Matrix")

File: 01-data-preprocessing/correlation_heatmap.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

df = pd.read_csv("movies.csv")
df = df.dropna(subset=["VOTES", "Gross"])
df["VOTES"] = df["VOTES"].str.replace(",", "").astype(int)
df["Gross"] = (df["Gross"].str.replace("$", "")).str.replace("M", "").astype(float)

numeric_df = df.select_dtypes(include=["float", "int"])
corr_matrix = numeric_df.corr()

sns.heatmap(corr_matrix, annot=True, fmt=".2f")
plt.title("Correlation Matrix")
plt.show()
